packages/@react-spectrum/dialog/src/DialogTrigger.tsx

```diff
@@ -64,37 +61,30 @@ function DialogTrigger(props: SpectrumDialogTriggerProps) {
     );
   }
 
-  let renderOverlay = () => {
-    switch (type) {
-      case 'fullscreen':
-      case 'fullscreenTakeover':
-        return (
-          <Modal isOpen={state.isOpen} isDismissable={false} onClose={state.close} type={type} isKeyboardDismissDisabled={isKeyboardDismissDisabled}>
-            {typeof content === 'function' ? content(state.close) : content}
-          </Modal>
-        );
-      case 'modal':
-        return (
-          <Modal isOpen={state.isOpen} isDismissable={isDismissable} onClose={state.close} isKeyboardDismissDisabled={isKeyboardDismissDisabled}>
-            {typeof content === 'function' ? content(state.close) : content}
-          </Modal>
-        );
-      case 'tray':
-        return (
-          <Tray isOpen={state.isOpen} onClose={state.close} isKeyboardDismissDisabled={isKeyboardDismissDisabled}>
-            {typeof content === 'function' ? content(state.close) : content}
-          </Tray>
-        );
-    }
-  };
+  const triggerBase = <DialogTriggerBase content={content}>{trigger}</DialogTriggerBase>;
 
+  // if container exists and not being used (showing an overlay), pass trigger base, otherwise
+  // create container for overlay to attach to
+  return dialogContainerContext && !dialogContainerContext.overlayContent ? triggerBase : (
```

Comment:
Ah I see. This might fix the nested overlay issue? I think this might be a little strange though. It means that where the overlay is mounted depends on whether another overlay is already open.